ui/brave_ads/message_popup_view_mac.mm

```diff
@@ -5,12 +5,27 @@
 
 #import "brave/ui/brave_ads/message_popup_view.h"
 
+#import <AppKit/AppKit.h>
 #import <Cocoa/Cocoa.h>
 
 #include "ui/views/widget/widget.h"
 
+namespace gfx {
+class Rect;
+}  // namespace gfx
+
 namespace brave_ads {
 
+gfx::Rect MessagePopupView::GetVisibleFrameForPrimaryDisplay() const {
+  NSRect frame = [[NSScreen mainScreen] frame];
+  NSRect visibleFrame = [[NSScreen mainScreen] visibleFrame];
+
+  // Normalize UI coordinate system origin from bottom-left to top-left
+  return gfx::Rect(visibleFrame.origin.x, frame.size.height -
```

Comment:
FYI - there is an api for converting this `gfx::ScreenRectFromNSRect()`.
and curious about that we need to calculate y by using frame and visibleFrame instead of visibleFrame's y?

Reply:
Ah cool, yes I will use this. The reason I had to have the calculation is on macOS 0, 0 is bottom-left of the screen. However this is now handled internally in `gfx::ScreenRectFromNSRect()`.